src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -444,7 +444,13 @@ void VM_Version::initialize() {
   }
 
   if (UseSVE > 0) {
-    _initial_sve_vector_length = get_current_sve_vector_length();
```

Comment:
We need an assert checking `_initial_sve_vector_length > 0`

Reply:
I don't like `get_current_sve_vector_length` returning `-1`. IMO if you cannot use SVE calling it makes no sense.
I see its doc mentions a negative result: https://github.com/openjdk/jdk/blob/b8233989e7605268dda908e6b639ca373789792b/src/hotspot/cpu/aarch64/vm_version_aarch64.hpp#L58-L62

Reply:
I don't quite get what are suggesting? AFAICS, we don't know we are in `prctl`-is-forbidden mess until we actually try to do it in `get_current_sve_vector_length()`. Then we have to deal with the consequences of `prctl` returning `-1`, which is what this PR does.

Reply:
> But I don't quite get what are you suggesting?

As the doc mentions a negative result, my suggestion to add an assert can be ignored.

Reply:
New commit adds more comprehensive checks.